lib/hanami/slice.rb

```diff
@@ -14,7 +14,7 @@ def initialize(application, name:, namespace: nil, root: nil, container: nil)
       @application = application
       @name = name.to_sym
       @namespace = namespace
-      @root = root
+      @root = root ? Pathname(root) : root
```

Comment:
I made it so that both the application and slices returned Pathnames for their roots. It's a useful thing to be able to rely on, particularly since we build on the root path so often.

(In this particular case, we do need to handle roots potentially being nil, since I designed slices so that – unlike applications – they don't necessarily _have_ to belong to a given directory)